Operations_on_polyhedra/include/CGAL/internal/Polyhedron_plane_clipping_3.h

```diff
@@ -241,28 +248,34 @@ Polyhedron* clip_polyhedron(Polyhedron& P, Polyhedron& clipping_polyhedron)
   return result.first;
 }
 
+
 template <class Polyhedron, class Plane_3>
 Polyhedron* clip_polyhedron(const Polyhedron& P, const Plane_3& p)
 {
-  CGAL::Bbox_3 bbox = CGAL::bbox_3(P.points_begin(), P.points_end());
-  //extend the bbox a bit to avoid border cases
-  double xd=(bbox.xmax()-bbox.xmin())/100;
-  double yd=(bbox.ymax()-bbox.ymin())/100;
-  double zd=(bbox.zmax()-bbox.zmin())/100;
-  bbox=CGAL::Bbox_3(bbox.xmin()-xd, bbox.ymin()-yd, bbox.zmin()-zd,
-                    bbox.xmax()+xd, bbox.ymax()+yd, bbox.zmax()+zd);
-  Polyhedron clipping_polyhedron=clip_bbox<Polyhedron>(bbox, p);
-
-  if (clipping_polyhedron.empty()) //no intersection, result is all or nothing
-  {
-    if (p.oriented_side(*P.points_begin())==ON_POSITIVE_SIDE)
-      return new Polyhedron();
-    else
-      return new Polyhedron(P);
+  if(! P.empty()){
+    CGAL::Bbox_3 bbox = CGAL::bbox_3(P.points_begin(), P.points_end());
+    //extend the bbox a bit to avoid border cases
```

Comment:
The function `bbox_3` should never return an uninitiated bounding box. If the range is empty, it could return the infinite bounding box, with ±∞ as coordinates.